receiver/awscontainerinsightreceiver/internal/k8sapiserver/k8sapiserver.go

```diff
@@ -279,6 +291,137 @@ func (k *K8sAPIServer) getReplicaSetMetrics(clusterName, timestampNs string) []p
 	return metrics
 }
 
+func (k *K8sAPIServer) getPodMetrics(clusterName, timestampNs string) []pmetric.Metrics {
+	var metrics []pmetric.Metrics
+	podsList := k.leaderElection.podClient.PodInfos()
+	podKeyToServiceNamesMap := k.leaderElection.epClient.PodKeyToServiceNames()
+
+	for _, podInfo := range podsList {
+		if podInfo.Phase == corev1.PodPending {
```

Comment:
Currently the api-server is only emitted pod status metrics for pending pods because the logic to extract the node name/instance name/instance type is complex.

I have kept the existing logic in the pod store file